src/mcode.ts

```diff
@@ -1178,74 +1193,76 @@ quantityMatch(
+    medication_mappings.set(['sacituzumab_govitecan_hziy'], ['sacituzumab_govitecan_hziy']);
+    medication_mappings.set(['methotrexate'], ['methotrexate']);
+    medication_mappings.set(['fluorouracil'], ['fluorouracil']);
+    medication_mappings.set(['vinorelbine'], ['vinorelbine']);
+    medication_mappings.set(['eribulin'], ['eribulin']);  // Originally spelled  eribuline, updated to eribulin.
+    medication_mappings.set(['etoposide'], ['etoposide']);
+    medication_mappings.set(['pemetrexed'], ['pemetrexed']);
+    medication_mappings.set(['irinotecan'], ['irinotecan']);
+    medication_mappings.set(['topotecan'], ['topotecan']);
+    medication_mappings.set(['ifosfamide'], ['ifosfamide']);
+    medication_mappings.set(['nivolumab'], ['nivolumab']);
+    medication_mappings.set(['avelumab'], ['avelumab']);
+    medication_mappings.set(['thiotepa'], ['thiotepa']);
+    medication_mappings.set(['olaparib'], ['olaparib']);
+    medication_mappings.set(['talazoparib'], ['talazoparib']);
+    medication_mappings.set(['atezolizumab'], ['atezolizumab']);
+    medication_mappings.set(['pembrolizumab'], ['pembrolizumab']);
+    medication_mappings.set(['zoledronic_acid'], ['zoledronic_acid']);
+    medication_mappings.set(['denosumab'], ['denosumab']);
+    medication_mappings.set(['bevacizumab'], ['bevacizumab']);
+    medication_mappings.set(['everolimus'], ['everolimus']);
+    medication_mappings.set(['progesterone'], ['progesterone']); // Originally spelled progestin, updated to progesterone.
+    medication_mappings.set(['fluoxymesterone'], ['fluoxymesterone']);
+    medication_mappings.set(['estrogen'], ['high_dose_estrogen']);  // Standard estrogen is the medication used for high_dose_estrogen, but Trialjectory expects high_dose_estrogen.
+    medication_mappings.set(['palbociclib'], ['palbociclib']);
+    medication_mappings.set(['abemaciclib'], ['abemaciclib']);
+    medication_mappings.set(['alpelisib'], ['alpelisib']);
+    medication_mappings.set(['ribociclib'], ['ribociclib']);
+    medication_mappings.set(['pertuzumab_trastuzumab_hyaluronidase'], ['pertuzumab_trastuzumab_hyaluronidase']);
+    medication_mappings.set(['goserelin'], ['goserelin']); // THIS MEDICATION IS NOT CURRENTLY SUPPORTED BY TRIALJECTORY. WE WILL NEED TO DISCUSS THIS WITH THEM.
+    medication_mappings.set(['leuprolide'], ['leuprolide']); // THIS MEDICATION IS NOT CURRENTLY SUPPORTED BY TRIALJECTORY. WE WILL NEED TO DISCUSS THIS WITH THEM.
 
     const medication_values: string[] = [];
 
-    // Iterate through the mappings and return when a code is satisfied.
-    for(const medication of medication_values_map.keys()){
-      console.log(medication)
-      if (this.cancerRelatedMedicationStatement.some((coding) => this.codeIsInSheet(coding, medication))) {
-        medication_values.push(...medication_values_map.get(medication));
+    // Iterate through the mappings.
+    for (const medication_sheets of medication_mappings.keys()) {
```

Comment:
This loop should be rewritten to loop through the bundle's medication statements, instead of looping through all of the possible medication mappings. Looping through the present medication statements, you should leverage the usage of the hashmap/dictionary and look up the medication in the mapping, not the mapping in the medications.

Reply:
Initial refactoring of the JSON mapping into a hashmap has been completed, holding off on this medication implementation because another PR has also refactored medications, want to prevent large merge conflict. Will address this once both PRs are merged, thanks! (see slack message)